deploy/helm/kubecf/templates/_config.tpl

```diff
@@ -0,0 +1,290 @@
+| _config.lookup (list $.kubecf $path)
++-----------------------------------------------------------------------------------------
+| Look up a config setting in $.kubecf.config at $path.
+|
+| $path can be a '.' separated string of property names 'features.eirini.enabled', a
+| list of names ("stacks" $stacks "releases"), or a combination of both. The '/' may
+| also be used in place of the '.' for a more JSON-patch look.
+|
+| See comments on the _config._lookup implementation function for more details.
+|
+| This function calls _config.load to make sure the config object is initialized.
+==========================================================================================
+*/}}
+{{- define "_config.lookup" }}
+  {{- $_ := include "_config.load" (first .) }}
+  {{- $kubecf := get (first .) "kubecf" }}
+  {{- $_ := set $kubecf "query" (join "." (rest .) | replace "/" ".") }}
+  {{- include "_config._lookup" (concat (list $kubecf $kubecf.config) (splitList "." $kubecf.query)) }}
+  {{- $_ := set $kubecf "required" false }}
+{{- end }}
+
+{{- /*
+==========================================================================================
+| _config.lookupRequired (list $.kubecf $path)
++-----------------------------------------------------------------------------------------
+| This function is currently unused and works identical to _config.lookup.
++-----------------------------------------------------------------------------------------
+| Not yet implemented:
+|
+| This function was supposed to be identical to _config.lookup, except it would fail
+| with a sensible error message if the $path did not exist. It may no longer be needed
+| now that we have JSON schema validation.
+==========================================================================================
+*/}}
+{{- define "kubecf.configRequired" }}
+  {{- $kubecf := get (first .) "kubecf" }}
+  {{- $_ := set $kubecf "required" true }}
+  {{- include "_config.lookup" . }}
+{{- end }}
+
+{{- /*
+==========================================================================================
+| _config.lookupManifest (list $.kubecf $path)
++-----------------------------------------------------------------------------------------
+| The same function as _config.lookup, except it searches $kubecf.manifest instead.
+==========================================================================================
+*/}}
+{{- define "_config.lookupManifest" }}
+  {{- $_ := include "_config.load" (first .) }}
+  {{- $kubecf := get (first .) "kubecf" }}
+  {{- $_ := set $kubecf "query" (join "." (rest .) | replace "/" ".") }}
+  {{- include "_config._lookup" (concat (list $kubecf $kubecf.manifest) (splitList "." $kubecf.query)) }}
+  {{- $_ := set $kubecf "required" false }}
+{{- end }}
+
+{{- /*
+==========================================================================================
+| _config._lookup (list $.kubecf $context $path)
```

Comment:
This basically exists to handle the `foo=bar` syntax right? On the happy path (with only dotted paths), it is essentially the same as normal property access?

Reply:
No, this historically existed because it would allow for throwing an understandable error message when the property did not exist. That code was not completely implemented, and has been removed for now.

So currently using `_config.lookup` on `$.Values` is mostly syntactic sugar over applying `index` repeatedly to walk down nested properties, and not throwing an error once an intermediate path doesn't exist.

But after the various refactorings and simplifications there are now instance where simple usage, like `{{ $releases := include "_config.lookup") (list $ "releases") }}` should be reverted back to `{{ $releases := $.Values.releases }}`.

I may clean up some at the end of my "implement code review feedback" cycle, but will leave others because I think I want to reintroduce the error reporting functionality in the future.

Reply:
I've replace all calls to `_config.lookup` with static parameter lists with a direct reference to `$.Values` instead in e1edd7a.

I've kept more complex queries like intact, e.g.

```
{{- $release_name := include "_config.lookup" (list $ "stacks" $stack_name "buildpacks" $buildpack_shortname) }}
```

These are the ones that would potentially benefit from proper error reporting, but again, that was planned before I added the schema validation; maybe that is no longer necessary either. But here the call also looks more readable than a series of nested `index` calls.